### DISCUSSION TITLE
Direct loading of .DMK as a TRS80 floppy image

### DISCUSSION THREAD

**Discussion by arekbr** [keirf/flashfloppy#728] [Ideas]
Good evening. 

I come with a question about a certain solution that would make TRS80 model III owners happy. Well, most of the floppy disk images for the TRS80 are in .DMK format. FlashFloppy does not load such an image. To be able to use the .DMK contents on a real TRS80 model III with a GOTEK hardware you have to use HxC soft to convert the .DMK to .jv3, and then this .jv3 to .HFE. Is there any chance that flashfloppy can use the .DMK format?

all the best
Arek

**Comment by arekbr** [comment]
Ehh... sorry. This thread came up in an issue, not a discussion of new ideas. I didn't come across it to check something that isn't there as an issue. My post can be deleted or ignored.  

All the best
Arek

**Comment by keirf** [comment]
There is a ticket from long ago, yes. I could give an update: Yes DMK is possible, but probably not on original STM and also AT32F415 Goteks due to lack of Flash space. This would probably be a feature for FlashFloppy Plus (ie. AT32F435 MCU). Read support would be quite easy. Write support a bit harder. Would write be needed? Just sector writes? Track format?

**Reply by arekbr** [reply]
For me, just reading the .DMK is enough. I have a TRS80 in my museum and I just want to load programmes when there are visitors. But now that you have written about the lack of space in the gotek, the issue seems completely irrelevant. I can take a moment and do the conversion from DMK to HFE. It's more of a whim :P than a feature I can't live without. Maybe actually, in the future, when more capacity can be had in gotek memory. But for now I find that it doesn't make sense after all. 

all the best
Arek

**Reply by keirf** [reply]
There are Goteks with more capacity, they have the new better MCU AT32F435: https://github.com/keirf/flashfloppy/wiki/Gotek-Models#microcontroller-options

**Reply by keirf** [reply]
But they aren't all that common, still.
